tasks/install_runner.yml

```diff
@@ -86,7 +86,9 @@
   become: yes
   become_user: "{{ runner_user }}"
   no_log: "{{ hide_sensitive_logs | bool }}"
-  when: runner_name not in registered_runners.json.runners|map(attribute='name')|list and not runner_org
+  when:
+   - runner_name not in registered_runners.json.runners|map(attribute='name')|list 
```

Comment:
I like the fact that you broken down the explicit `and`s into lists, but I feel there is a lot repeated code here and its doing my head in to look for the `in` and `not in`.

Could we:
* contain the logic in some facts
* use block to group similarly conditioned tasks

```
  set_fact:
     is_registered_runner: "{{ runner_name n registered_runners.json.runners|map(attribute='name')|list }}"

...

- name: Register a new service
  when: not is_registered_runner
  block:
    - name: Register runner (if new installation) for repository
       ...
      when: runner_type == "repository"

    ...

- name: Replace registered runner
  when:
    - is_registered_runner
    - reinstall_runner
      
  ...
```

Reply:
Here I was mirroring the practices in the upstream repo.
It's annoying to read when the tasks aren't blocked together, but I think we're better to leave it that way.

Reply:
From Stephen Thirlwwall:
> If I get a drive-by PR with some useful feature and they've taken it upon themselves to reorganise/refactor large chunks of the code, I'm far less likely to accept that PR.
> I'd be inclined to make it a two parter. First implement your feature using their style. Then if that gets accepted, make another PR with the whole lot refactored into blocks.

I think what he has said makes sense so, list leave it as is, For this PR.